R/plot-construction.R

```diff
@@ -87,49 +87,80 @@ add_ggplot <- function(p, object, objectname) {
 #'
 #' @keywords internal
 #' @export
-ggplot_add <- function(object, plot, object_name) {
-  UseMethod("ggplot_add")
-}
-#' @export
-ggplot_add.default <- function(object, plot, object_name) {
+ggplot_add <- new_generic(
+  "ggplot_add",
+  dispatch_args = c("object", "plot"),
+  fun = function(object, plot, object_name) S7_dispatch()
+)
+
+class_ggplot <- new_S3_class("ggplot")
+
+method(
+  ggplot_add,
+  list(object = class_any, plot = class_ggplot)
+) <- function(object, plot, object_name) {
```

Comment:
I think it's easier to read on one line:

```suggestion
method(ggplot_add, list(class_any, class_ggplot)) <- function(object, plot, object_name) {
```

Reply:
This makes sense to me, but some lines became quite long, so I declared the function on a new line. I don't think the tidyverse style guide has explicit guidance on long LHS assignments.